docs/docsite/_ext/ansible_redirects.py

```diff
@@ -0,0 +1,27 @@
+"""Sphinx extension for generating redirects in Ansible package docs."""
+
+import os
+from pathlib import Path
+
+import yaml
+from sphinx.application import Sphinx
+
+DOCSITE_ROOT_DIR = Path(__file__).parents[1].resolve()
+ANSIBLE_REDIRECTS_CONFIG_PATH = DOCSITE_ROOT_DIR / "ansible_redirects.yaml"
+REDIRECT_KEYS = yaml.safe_load(
+    ANSIBLE_REDIRECTS_CONFIG_PATH.read_text(encoding="utf-8")
```

Comment:
Let's not do any I/O during import time and move it to runtime behind `setup()` or further into some lifecycle hook.